pkg/operator/ceohelpers/unsupported_override.go

```diff
@@ -0,0 +1,29 @@
+package ceohelpers
+
+import (
+	"bytes"
+	"encoding/json"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+)
+
+// IsUnsupportedUnsafeEtcd returns true if
+// useUnsupportedUnsafeNonHANonProductionUnstableEtcd key is set
+// to any parsable value
+func IsUnsupportedUnsafeEtcd(spec *operatorv1.StaticPodOperatorSpec) (bool, error) {
+	unsupportedConfig := map[string]interface{}{}
+	if spec.UnsupportedConfigOverrides.Raw == nil {
+		return false, nil
+	}
+	if err := json.NewDecoder(bytes.NewBuffer(spec.UnsupportedConfigOverrides.Raw)).Decode(&unsupportedConfig); err != nil {
+		klog.V(4).Infof("decode of unsupported config failed with error: %v", err)
+		return false, err
+	}
+
+	_, found, err := unstructured.NestedFieldNoCopy(unsupportedConfig, "useUnsupportedUnsafeNonHANonProductionUnstableEtcd")
```

Comment:
Users hate having bool-ish field not act like bools.  Check for truthy-ness

Reply:
@hexfusion was concerned if we would panic if a user does a string `"true"` in the yaml instead of `bool` hence added test cases to make sure we consider both bool and string.